charts/endeavor/templates/_environment.tpl

```diff
@@ -0,0 +1,58 @@
+{{/*
+Endeavor pods are configured primarily through the environment. Environment variables
+from the values.yaml file are defined here and provided as configuration to the pod.
+*/}}
+{{- define "endeavor.environment" -}}
+env:
+  - name: ENDEAVOR_MAINTENANCE
+    value: {{ .Values.endeavor.maintenance | quote }}
+  - name: ENDEAVOR_MODE
+    value: {{ .Values.endeavor.mode | quote }}
+  - name: ENDEAVOR_LOG_LEVEL
+    value: {{ .Values.endeavor.logLevel | quote }}
+  - name: ENDEAVOR_CONSOLE_LOG
+    value: {{ .Values.endeavor.consoleLog | quote }}
+  - name: ENDEAVOR_DATABASE_URL
+    value: {{ .Values.endeavor.databaseURL | quote }}
+  - name: ENDEAVOR_DOCUMENTS_URL
+    value: {{ .Values.endeavor.documentsURL | quote }}
+  - name: ENDEAVOR_WEB_MAINTENANCE
+    value: {{ .Values.endeavor.web.maintenance | quote }}
+  - name: ENDEAVOR_WEB_ENABLED
+    value: {{ .Values.endeavor.web.enabled | quote }}
+  - name: ENDEAVOR_WEB_API_ENABLED
+    value: {{ .Values.endeavor.web.APIenabled | quote }}
+  - name: ENDEAVOR_WEB_UI_ENABLED
+    value: {{ .Values.endeavor.web.UIenabled | quote }}
+  - name: ENDEAVOR_WEB_BIND_ADDR
+    value: {{ .Values.endeavor.web.bindAddr | quote }}
+  - name: ENDEAVOR_WEB_ORIGIN
+    value: {{ .Values.endeavor.web.origin | quote }}
```

Comment:
This value should be created from the service/port info in `.values.yaml` rather than from `endeavor.web.origin`.